subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/TaskFromPluginValidationIntegrationTest.groovy

```diff
@@ -0,0 +1,179 @@
+
+    def setup() {
+        expectReindentedValidationMessage()
+    }
+
+    @ValidationTestFor(
+        ValidationProblemId.TEST_PROBLEM
+    )
+    def "detects that a problem is from a task declared in a precompiled script plugin"() {
+        withPrecompiledScriptPlugins()
+        def pluginFile = file("buildSrc/src/main/groovy/org.gradle.demo.plugin.gradle")
+        writeTaskInto(pluginFile)
+        pluginFile << """
+            tasks.register("myTask", SomeTask) {
+                input.set("hello")
+                output.set(layout.buildDirectory.file("out.txt"))
+            }
+        """
+
+        buildFile """plugins {
+            id 'org.gradle.demo.plugin'
+        }"""
+
+        when:
+        fails ':myTask'
+
+        then:
+        failureDescriptionContains(dummyValidationProblem {
+            inPlugin('org.gradle.demo.plugin')
+            type('SomeTask').property('input')
+        }.trim())
+    }
+
+    @ValidationTestFor(
+        ValidationProblemId.TEST_PROBLEM
+    )
+    def "detects that a problem is from a task declared in plugin"() {
+        settingsFile << """
+            includeBuild 'my-plugin'
+        """
+        copyValidationProblemClass()
+        def pluginFile = file("my-plugin/src/main/groovy/org/gradle/demo/plugin/MyTask.groovy")
+        writeTaskInto("""package org.gradle.demo.plugin
+
+            import org.gradle.api.*
+            import org.gradle.api.file.*
+            import org.gradle.api.provider.*
+            import org.gradle.api.tasks.*
+        """, pluginFile)
+
+        file("my-plugin/build.gradle") << """
+            plugins {
+                id 'groovy-gradle-plugin'
+            }
+
+            gradlePlugin {
+                plugins {
+                    create("simplePlugin") {
+                        id = "org.gradle.demo.plugin"
+                        implementationClass = "org.gradle.demo.plugin.MyPlugin"
+                    }
+                }
+            }
+        """
+        file("my-plugin/settings.gradle") << """
+            rootProject.name = "my-plugin"
+        """
+        file("my-plugin/src/main/groovy/org/gradle/demo/plugin/MyPlugin.groovy") << """package org.gradle.demo.plugin
+            import org.gradle.api.*
+            class MyPlugin implements Plugin<Project> {
+                void apply(Project p) {
+                    p.tasks.register("myTask", SomeTask) {
+                        it.input.set("hello")
+                        it.output.set(p.layout.buildDirectory.file("out.txt"))
+                    }
+                }
+            }
+        """
+
+        buildFile """plugins {
+            id 'org.gradle.demo.plugin'
+        }"""
+
+        when:
+        fails ':myTask'
+
+        then:
+        failureDescriptionContains(dummyValidationProblem {
+            inPlugin('org.gradle.demo.plugin')
+            type('SomeTask').property('input')
+        }.trim())
```

Comment:
Why do we need to call `trim()` here, shouldn't the test fixture take care of this?

Reply:
fixed